python/pyspark/sql/connect/dataframe.py

```diff
@@ -1836,11 +1839,26 @@ def _schema(self) -> StructType:
         if self._cached_schema is None:
             query = self._plan.to_proto(self._session.client)
             self._cached_schema = self._session.client.schema(query)
+            try:
+                self._cached_schema_serialized = CPickleSerializer().dumps(self._schema)
+            except Exception as e:
+                logger.warn(f"DataFrame schema pickle dumps failed with exception: {e}.")
+                self._cached_schema_serialized = None
         return self._cached_schema
 
     @property
     def schema(self) -> StructType:
-        return copy.deepcopy(self._schema)
+        copied_schema: Optional[StructType] = None
+        if self._cached_schema_serialized is not None:
+            try:
+                copied_schema = CPickleSerializer().loads(self._cached_schema_serialized)
+            except Exception as e:
+                logger.warn(f"DataFrame schema pickle loads failed with exception: {e}.")
+                copied_schema = None
+        if copied_schema:
+            return copied_schema
+        else:
+            return copy.deepcopy(self._schema)
```

Comment:
More idiomatic version.
```suggestion
        if self._cached_schema_serialized is not None:
            try:
                return CPickleSerializer().loads(self._cached_schema_serialized)
            except Exception as e:
                logger.warn(f"DataFrame schema pickle loads failed with exception: {e}.")
        return copy.deepcopy(self._schema)
```